src/libraries/System.Net.Sockets/tests/FunctionalTests/DisposedSocketTests.cs

```diff
@@ -752,13 +752,14 @@ public void EndAccept_Throws_ObjectDisposed()
         [InlineData(true)]
         public async Task NonDisposedSocket_SafeHandlesCollected(bool clientAsync)
         {
-            List<WeakReference> handles = await CreateHandlesAsync(clientAsync);
-            RetryHelper.Execute(() =>
+            TimeSpan timeout = TimeSpan.FromSeconds(10);
+            List<WeakReference> handles = await CreateHandlesAsync(clientAsync).WaitAsync(timeout);
+            await RetryHelper.ExecuteAsync(() => Task.Run(() =>
             {
                 GC.Collect();
                 GC.WaitForPendingFinalizers();
                 Assert.Equal(0, handles.Count(h => h.IsAlive));
-            });
+            })).WaitAsync(timeout);
```

Comment:
Will RetryHelper quit after a fixed number of retries? If not, you will end up having this run forever in the background, which seems bad...

Reply:
Yes, the default number of retries is around 10, but I don't think it helps if any of those runs hang.

In case of timeout, `WaitAsync` will just abandon the original `Task`, which will keep running forever in the background, which is indeed bad, but IMO still much better than hanging the test execution without auditable results.

Reply:
Yeah, that's fine. I just wanted to make sure we weren't going to keep forcing GC over and over without end.